### PR TITLE
fix: fix `accountCreated` handling, add `storageTrace`, fix `DELEGATECALL`/`STATICCALL` caller state

### PR DESCRIPTION
**Pull request by noel2004**
TODO：update zkevm-circuits

As discussed in the issue. Some inconsistecies are caused by unsuitable initialization of stateDB used by busmapping. This PR change the initialization by adding states being traced in `storageTrace` field of block results and this work should fix most of the inconsistency.

This PR also change the example traces by manually revert the byte order in proofs, because they [have been unified into big-endian](https://github.com/scroll-tech/go-ethereum/pull/133).

### CODE REVIEW

**Comment by noel2004** [comment]
I have test this PR on the skipped blocks in roller and all of them can passed the handl-block phase (so witness can be assigned and circuit be built), execpt for block 17, which assert with `!callee_code_hash.is_zero()`.

**Comment by icemelon** [comment]
@noel2004 I checked the transaction in the block 17 and it calls to the L2Messenger contract. The contract shouldn't have a zero codehash. So not sure why the assert would fail for this block.

**Comment by lispc** [comment]
You can Leave this error case to me. You can really proved some blocks in our dev server (i suggest only prove evm circuit. Don't prove agg)

**Comment by lispc** [comment]
Suggest Set MOCK_PROVE env var to true to get better circuit error msg.

**Comment by noel2004** [comment]
Now integeration test on evm circuit has passed for following blocks:
4, 6, 10, 14, 17, 19, 21

**Comment by HAOYUatHZ** [comment]
I will build and test without merging first.

BTW what dores `issue.json` mean? 
1. it's not block 17.
2. issue #36? but there're 4 problems there.

File: types/src/eth.rs [hunk 1]
```diff
@@ -23,6 +24,8 @@ pub struct BlockResult {
     pub execution_results: Vec<ExecutionResult>,
     #[serde(rename = "mptwitness", skip_serializing, default)]
     pub mpt_witness: Vec<SMTTrace>,
+    #[serde(rename = "storageTrace", skip_serializing)]
```
Comment (HAOYUatHZ) [inline types/src/eth.rs:27]: should we
```suggestion
    #[serde(rename = "storageTrace", skip_serializing, default)]
```
？

Comment (noel2004) [reply types/src/eth.rs:27]: I think `storageTrace` has been definitely a field existed in blockResult so without `default` instruction we could detect some irregulation early.

Comment (HAOYUatHZ) [reply types/src/eth.rs:27]: idk.

I proposed this becaseu I encountered an error here: https://github.com/scroll-tech/roller-rs/pull/165

```
thread 'types::test::codec_traces' panicked at 'called `Result::unwrap()` on an `Err` value: Error("missing field `storageTrace`", line: 1, column: 430)', src/types.rs:79:76
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test types::test::codec_traces ... FAILED
[0m[91m   Compiling pest_generator v2.1.3
thread 'queue::test::fifo' panicked at 'called `Result::unwrap()` on an `Err` value: Error("missing field `storageTrace`", line: 1, column: 325)', src/queue.rs:40:84
test queue::test::fifo ... FAILED
```


There might be solution but I haven't took a close look

**Comment by noel2004** [comment]
> I will build and test without merging first.
> 
> BTW what dores `issue.json` mean?
> 
> 1. it's not block 17.
> 2. issue [inconsistency between state rebuilding #36](https://github.com/scroll-tech/common-rs/issues/36)? but there're 4 problems there.

issue.json is being added accidentally. It would be prune

**Comment by noel2004** [comment]
Now the PR has passed with block 17 so it should be ready.

Work left:

+ lacking of serialize in storeTrace may cause roller fail for its store / reloading process.

File: types/src/eth.rs (hunk 1 above)
Comment (noel2004) [reply types/src/eth.rs]: Seralize feature has been added. This issue should be resolved.

**Comment by noel2004** [comment]
Now the `blockResult` structure can be fully serialized / deserialized.

**Review by HAOYUatHZ — APPROVED** [review]

**Pull request merged**